README.md

```diff
@@ -745,6 +745,188 @@ This section provides examples of interacting with the Envoy External Authorizat
+* Envoy Version: 1.10.0
+* OPA-Envoy Version: 0.26.0-envoy
+
+### Benchmarks
+
+The benchmark result below provides the percentile distribution of the latency observed by sending *100 requests/sec*
+to the sample application. Each request makes a `GET` call to the `/people` endpoint exposed by the application.
+
+The graph shows the latency distribution when the load test is performed under the following conditions:
+
+1. **App Only**
+
+In this case, the graph documents the latency distribution observed when requests are
+sent directly to the application ie. no Envoy and OPA in the request path. This scenario is depicted by the
+`blue` curve.
+
+2. **App and Envoy**
+
+In this case, the distribution is with [Envoy External
+Authorization
+API](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ext_authz_filter.html) disabled. This means
+OPA is not included in the request path but Envoy is. This scenario is depicted by the `red` curve.
+
+3. **App, Envoy and OPA (NOP policy)**
+
+In the case, we will see the latency observed with [Envoy External
+Authorization
+API](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ext_authz_filter.html) enabled. This means
+Envoy will make a call to OPA on every incoming request. The graph explores the effect of loading the below NOP policy into
+OPA. This scenario is depicted by the `green` curve.
+
+```rego
+package envoy.authz
+
+default allow = true
+```
+
+4. **App, Envoy and OPA (RBAC policy)**
+
+In the case, we will see the latency observed with [Envoy External
+Authorization
+API](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ext_authz_filter.html) enabled and
+explore the effect of loading the following RBAC policy into OPA. This scenario is depicted by the `yellow` curve.
+
+```rego
+package envoy.authz
+
+import input.attributes.request.http as http_request
+
+default allow = false
+
+allow {
+    roles_for_user[r]
+    required_roles[r]
+}
+
+roles_for_user[r] {
+    r := user_roles[user_name][_]
+}
+
+required_roles[r] {
+    perm := role_perms[r][_]
+    perm.method = http_request.method
+    perm.path = http_request.path
+}
+
+user_name = parsed {
+    [_, encoded] := split(http_request.headers.authorization, " ")
+    [parsed, _] := split(base64url.decode(encoded), ":")
+}
+
+user_roles = {
+    "alice": ["guest"],
+    "bob": ["admin"]
+}
+
+role_perms = {
+    "guest": [
+        {"method": "GET",  "path": "/people"},
+    ],
+    "admin": [
+        {"method": "GET",  "path": "/people"},
+    ],
+}
+```
+
+![hist_100](./docs/hist_google_grpc_100.png)
+
+The above four scenarios are replicated to measure the latency distribution now by sending *1000 requests/sec*
+to the sample application. The following graph captures this result.
+
+![hist_1000](./docs/hist_google_grpc_1000.png)
+
+#### OPA Benchmarks
+
+The table below captures the `gRPC Server Handler` and `OPA Evaluation` time with [Envoy External
```

Comment:
Do we explain what these two components mean anywhere? It's probably worth mentioning that the `gRPC Server Handler` includes `OPA Evaluation`. In other words, `OPA Evaluation` is the time taken to evaluate the policy. `gRPC Server Handler` is the total time taken to prepare the input for the policy, evaluate the policy (`OPA Evaluation`), and prepare the result.

Reply:
Done